sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -288,37 +310,60 @@ public boolean isDisposed() {
         return isDisposed.get();
     }
 
+    /**
+     * Blocking call that disposes of the sender. See {@link #dispose(String, ErrorCondition)}.
+     */
     @Override
     public void dispose() {
-        dispose(null);
+        dispose("Dispose called", null)
+            .publishOn(Schedulers.boundedElastic())
+            .block(retryOptions.getTryTimeout());
     }
 
     /**
-     * Disposes of the sender when an exception is encountered.
+     * Disposes of the sender.
      *
      * @param errorCondition Error condition associated with close operation.
+     * @param message Message associated with why the sender was closed.
+     *
+     * @return A mono that completes when the send link has closed.
      */
-    void dispose(ErrorCondition errorCondition) {
+    Mono<Void> dispose(String message, ErrorCondition errorCondition) {
```

Comment:
Dispose is asynchronous because when we change `sender.close()`, it will set the _local_ state of the send link. Then we need the remote to also acknowledge it via the `onLinkRemoteClosed` event.